### PR TITLE
feat: Genesis Setters

### LINKED ISSUE #1631: Genesis Helper methods
### Describe the feature

Helper methods for creating the primitive types `Genesis` and `GenesisAccount`. Currently, there is no convenient way to build `ChainSpec`s other than mainnet, sepolia, and goerli.

It would be nice to initialize these structs using helper methods like [`PeersConfig`](https://github.com/paradigmxyz/reth/blob/main/crates/net/network/src/peers/manager.rs#L991-L1055), [`RpcServerConfig`](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-builder/src/lib.rs#L670-L715), and [`RpcModuleBuilder`](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-builder/src/lib.rs#L152-L186) to name a few.

### Additional context

Local development chains can be more easily configured using these helper methods like so:
```
fn build_chain_spec() -> ChainSpec {
    let chain = Chain::Named(ethers_core::types::Chain::Dev);

    let custom_genesis = Genesis::default()
        .with_nonce(nonce)
        .with_gas_limit(gas_limit)
        .with_difficulty(difficulty)
        .with_mix_hash(mix_hash)
        .with_accounts(genesis_accounts);

    let chain_spec = ChainSpecBuilder::default()
        .chain(chain)
        .genesis(custom_genesis)
        .build();

    chain_spec
}
```

### PR DESCRIPTION
**Pull request by grantkee**
Helper methods for `Genesis` and `GenesisAccount` with unit tests

I decided to prefix the methods using `with_` instead of `set_` because it seemed more consistent with the project. I avoided creating `GenesisBuilder` structs because these genesis structs seem more align with [`PeerConfig`](https://github.com/paradigmxyz/reth/blob/main/crates/net/network/src/peers/manager.rs#L991-L1055) than the RPC module structs.

If you think it would be more appropriate to have a builder, I can refactor.

closes #1631

### CODE REVIEW

File: crates/primitives/src/genesis.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use serde::{Deserialize, Serialize};
 use triehash::sec_trie_root;
 
 /// The genesis block specification.
-#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
```
Comment (mattsse) [inline crates/primitives/src/genesis.rs:15]:
```suggestion
#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
```

File: crates/primitives/src/genesis.rs [hunk 2]
```diff
@@ -36,6 +36,57 @@ pub struct Genesis {
...
+
+    /// Set the coinbase address.
+    pub fn with_coinbase(mut self, address: Address) -> Self {
+        self.coinbase = address;
+        self
+    }
+
+    /// Add accounts to the genesis block. If the address is already present,
+    /// the account is updated.
+    pub fn with_accounts(mut self, accounts: HashMap<Address, GenesisAccount>) -> Self {
```
Comment (mattsse) [inline crates/primitives/src/genesis.rs:84]:
```suggestion
    pub fn extend_accounts(mut self, accounts: impl IntoIter<Item = (Address, GenesisAccount)>) -> Self {
```

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nits, otherwise lgtm

**Comment by grantkee** [comment]
@mattsse Thanks for the feedback!

**Review by gakonst — APPROVED** [review]

**Pull request merged**